utils/expire_map.go

```diff
@@ -75,7 +77,8 @@ func (e *ExpiredMap) Get(key interface{}) (interface{}, bool) {
 		}
 
 		// Cache expires, updated via updateHandler.
-		if e.UpdateHandler != nil {
+		// Check eval.updated to avoid cache flood.
+		if e.UpdateHandler != nil && atomic.CompareAndSwapUint32(&eval.updated, 0, 1) {
```

Comment:
when to set updated to 0?

Reply:
The `Set`  and `e.updateData` function would create new  `expiredata` and `updated` default value is 0.

Reply:
Need fix when update failed.